crates/router/src/routes/api_keys.rs

```diff
@@ -66,9 +63,9 @@ pub async fn api_key_create(
 pub async fn api_key_retrieve(
     state: web::Data<AppState>,
     req: HttpRequest,
-    path: web::Path<String>,
+    path: web::Path<(String, String)>,
 ) -> impl Responder {
-    let key_id = path.into_inner();
+    let (key_id, _merchant_id) = path.into_inner();
```

Comment:
Won't the `key_id` and `merchant_id` variables have interchanged values? Same with update and revoke endpoints.